CONTRIBUTING.md

```diff
@@ -132,24 +132,92 @@ In order to build and publish ODK, you need the following:
+* By convention, changes to the next major branch that are introduced by a PR tagged with a `hotfix` label should also be backported to the maintenance branch.
+* To avoid cluttering the maintenance branch with multiple “Python constraints update” backport commits, it is recommended to backport all Python constraints at once, shortly before a minor release.
+* There are no strict guidelines about when a minor release should happen. The availability of a new version of ROBOT is usually reason enough to make such a release, but upgrades to other tools can also occasionally justify a minor release.
+
+Once the decision to make a minor release has been made:
+
+* Make sure all tools-related updates (including Python tools) have been backported.
+* Do any amount of testing as needed to be confident we are ready for release. For minor releases, it makes sense to test the ODK on at least 5 ontologies. In 2024 we typically test:
+  - [Mondo](https://github.com/monarch-initiative/mondo) ([docs](https://mondo.readthedocs.io/en/latest/developer-guide/release/)) (a lot of use of old tools, like owltools, interleaved with ROBOT, heavy dependencies on serialisations, perl scripts)
+  - [Mondo Ingest](https://github.com/monarch-initiative/mondo-ingest) (a lot of use of sssom-py and OAK, interleaved with heavyweight ROBOT pipelines)
+  - [Uberon](https://github.com/obophenotype/uberon) (ROBOT plugins, old tools like owltools)
+  - [Human Phenotype Ontology](https://github.com/obophenotype/human-phenotype-ontology) (uses of ontology translation system (babelon), otherwise pretty standard ODK, high impact ontology)
+  - [Cell Ontology](https://github.com/obophenotype/cell-ontology) (Relatively standard, high impact ODK setup)
+* Update the CHANGELOG.md file.
+* Bump the version number to `v1.X.Y` in
+  - the top-level `Makefile`,
+  - the `Makefile` in the `docker/odklite` directory.
+* If the minor release includes a newer version of ROBOT, and if that has not already been done when ROBOT itself was updated, update the version number in `docker/robot/Makefile` so it matches the version of ROBOT that is used.
+* Push all last-minute changes (CHANGELOG and version number updates) to the `BRANCH-1.X-MAINTENANCE` branch.
+* Build and publish the images from the tip of the `BRANCH-1.X-MAINTENANCE` branch (same procedure as above to build and publish a major release).
+* Create a GitHub release from the tip of the `BRANCH-1.X-MAINTENANCE` branch, with a `v1.X.Y` tag.
+* Resume backporting changes to the `BRANCH-1.X-MAINTENANCE` until the time comes for the next minor release.
+
+### Development snapshot releases
+
+Development snapshot releases reflect the current state of the `main` (`master`) branch. They are routine releases where we do not do a lot of testing, and anyone using these
+do so at their own risk.
+The ODK team is not accountable for pipelines that fail because of changes to the development snapshot.
```

Comment:
I’d rather not call them “routine releases”. I guess it’s subject to interpretation, but to me, saying that something is done “routinely” kind of implies a certain form a stability, as in “it’s all routine stuff, there’s nothing surprising happening in those releases, and therefore nothing that could break your workflows” – exactly the opposite of what the `-dev` ”releases” are!

In fact I would not call them “releases” at all. They are “development snapshots”, not “development snapshot releases”.

My suggestion:

```
### Development snapshots

Development snapshots reflect the current state of the main (`master`) branch. They do not undergo the same level of testing (or any testing at all) as the normal releases, and are intended to help trialing and debugging the changes that happen in the `master` branch.

Development snapshots should _not_ be used in a production environment. Feel free to use them if you want to help us developing the next major release, but if you use them in your production pipelines, understand that you’re doing so at your own risk.

Reply:
Makes sense. Done!